File: mobile/src/main/java/jp/satorufujiwara/kotlin/AppGlideModule.java
package jp.satorufujiwara.kotlin;

import com.bumptech.glide.Glide;
import com.bumptech.glide.GlideBuilder;
import com.bumptech.glide.integration.okhttp.OkHttpUrlLoader;
import com.bumptech.glide.load.DecodeFormat;
import com.bumptech.glide.load.engine.cache.LruResourceCache;
import com.bumptech.glide.load.model.GlideUrl;
import com.bumptech.glide.module.GlideModule;
import com.squareup.okhttp.OkHttpClient;

import android.content.Context;

import java.io.InputStream;

public class AppGlideModule implements GlideModule {

    @Override
    public void applyOptions(Context context, GlideBuilder builder) {
        builder.setMemoryCache(new LruResourceCache(30 * 1024 * 1024))
                .setDecodeFormat(DecodeFormat.PREFER_ARGB_8888);
    }

    @Override
    public void registerComponents(Context context, Glide glide) {
        //no op
    }

    public static void registerComponents(Context context, OkHttpClient okHttpClient) {
        Glide.get(context).register(GlideUrl.class, InputStream.class,
                new OkHttpUrlLoader.Factory(okHttpClient));
    }
}

File: mobile/src/main/kotlin/jp/satorufujiwara/kotlin/AppGlideModule.kt
package jp.satorufujiwara.kotlin

import android.content.Context
import com.bumptech.glide.Glide
import com.bumptech.glide.GlideBuilder
import com.bumptech.glide.integration.okhttp.OkHttpUrlLoader
import com.bumptech.glide.load.DecodeFormat
import com.bumptech.glide.load.engine.cache.LruResourceCache
import com.bumptech.glide.load.model.GlideUrl
import com.bumptech.glide.module.GlideModule
import com.squareup.okhttp.OkHttpClient
import java.io.InputStream

class AppGlideModule : GlideModule {

    override fun applyOptions(context: Context, builder: GlideBuilder) {
        builder.setMemoryCache(LruResourceCache(30 * 1024 * 1024)).setDecodeFormat(
                DecodeFormat.PREFER_ARGB_8888)
    }

    override fun registerComponents(context: Context, glide: Glide) {
        //no op
    }

    companion object {

        fun registerComponents(context: Context, okHttpClient: OkHttpClient) {
            Glide.get(context).register(GlideUrl::class.java, InputStream::class.java,
                    OkHttpUrlLoader.Factory(okHttpClient))
        }
    }
}
